### ISSUE TITLE
'CogVideoXTransformer3DModel' object has no attribute 'context_embedder'

### ISSUE THREAD

**Issue by Kh4rj1** [kijai/ComfyUI-CogVideoXWrapper#101]
I've been trying to run CogVideo locally using multiple different worklows but I always run into the same error sooner or later, it's like the final "barrier" I can't get past. When the program comes to the "(down)load CogVideo Model) Node, I get an error saying 'CogVideoXTransformer3DModel' object has no attribute 'context_embedder'

full error log here: https://pastebin.com/B6fvHq4v

The workflow I'm using: https://github.com/henrique-galimberti/i2v-workflow/blob/main/CogVideoX-I2V-workflow.json

I'm kind of at my wits end, i tried updating everything that could be updated but I still can't get past this

**Comment by kijai** [comment]
Well... CogVideoX indeed doesn't have `context_embedder`, from what I can see that's only a thing for Flux and SD3 models. So I don't know why you would be getting that error.

All I can think of is to double check your diffusers version is 0.30.3, and maybe also update accelerate to latest.

And be sure to use the python.exe in your ComfyUI\python_embded -folder:

`python.exe -m pip install -U diffusers accelerate`

If that doesn't help, then similarly in that folder do:

`python.exe -m pip list` and show me the results for further troubleshooting.

**Comment by Kh4rj1** [comment]
I ran the command as you said but unfortunately the issue persist.

Here's the list:
`C:\AI\ComfyUI\python_embeded>python.exe -m pip list
Package                           Version
--------------------------------- ---------------
absl-py                           2.1.0
accelerate                        0.34.2
addict                            2.4.0
aiohappyeyeballs                  2.4.2
aiohttp                           3.10.7
aioresponses                      0.7.6
aiosignal                         1.3.1
albucore                          0.0.17
albumentations                    1.4.16
annotated-types                   0.7.0
anthropic                         0.34.2
antlr4-python3-runtime            4.9.3
anyio                             4.6.0
APScheduler                       3.10.1
arrow                             1.3.0
attrs                             23.2.0
auto_gptq                         0.7.1
awscli                            1.33.44
babel                             2.16.0
backoff                           2.2.1
beautifulsoup4                    4.12.3
binaryornot                       0.4.4
bitsandbytes                      0.44.0
blend_modes                       2.1.0
blind-watermark                   0.4.4
boto3                             1.34.123
botocore                          1.34.162
cachetools                        5.5.0
certifi                           2024.7.4
cffi                              1.16.0
chardet                           5.2.0
charset-normalizer                3.3.2
click                             8.1.7
clip-interrogator                 0.6.0
cmake                             3.30.3
color-matcher                     0.5.0
colorama                          0.4.4
coloredlogs                       15.0.1
colour-science                    0.4.4
contourpy                         1.3.0
cookiecutter                      2.6.0
cryptography                      43.0.0
cstr                              0.1.0
cupy-cuda12x                      13.3.0
cycler                            0.12.1
Cython                            3.0.0
dataclasses-json                  0.6.7
datasets                          3.0.1
ddt                               1.7.2
decorator                         4.4.2
defusedxml                        0.7.1
Deprecated                        1.2.14
diffusers                         0.30.3
dill                              0.3.8
diskcache                         5.6.3
distro                            1.9.0
docker                            6.1.3
docker-pycreds                    0.4.0
docstring_parser                  0.16
docutils                          0.16
easydict                          1.13
einops                            0.8.0
eval_type_backport                0.2.0
fairscale                         0.4.0
fal_client                        0.4.1
fastapi                           0.110.3
fastrlock                         0.8.2
ffmpy                             0.3.0
filelock                          3.15.4
flatbuffers                       24.3.25
flet                              0.24.1
flet-core                         0.24.1
flet-runtime                      0.24.1
fonttools                         4.54.1
frozenlist                        1.4.1
fsspec                            2024.6.1
ftfy                              6.2.3
gdown                             5.2.0
gekko                             1.2.1
ghp-import                        2.1.0
gitdb                             4.0.11
GitPython                         3.1.43
google-ai-generativelanguage      0.6.10
google-api-core                   2.20.0
google-api-python-client          2.147.0
google-auth                       2.35.0
google-auth-httplib2              0.2.0
google-cloud-core                 2.4.1
google-cloud-storage              2.18.2
google-crc32c                     1.6.0
google-generativeai               0.8.2
google-resumable-media            2.7.2
googleapis-common-protos          1.65.0
GPUtil                            1.4.0
griffe                            1.3.1
grpcio                            1.66.2
grpcio-status                     1.66.2
h11                               0.14.0
httpcore                          1.0.5
httplib2                          0.22.0
httptools                         0.6.1
httpx                             0.27.2
httpx-sse                         0.4.0
huggingface-hub                   0.24.2
humanfriendly                     10.0
hydra-core                        1.3.2
idna                              3.7
imageio                           2.35.1
imageio-ffmpeg                    0.5.1
img2texture                       1.0.6
importlib_metadata                8.5.0
inference-cli                     0.21.0
inference-gpu                     0.21.0
iniconfig                         2.0.0
intel-openmp                      2021.4.0
jax                               0.4.33
jaxlib                            0.4.33
Jinja2                            3.1.4
jiter                             0.5.0
jmespath                          1.0.1
joblib                            1.4.2
jsonschema                        4.23.0
jsonschema-specifications         2023.12.1
kiwisolver                        1.4.7
kornia                            0.7.3
kornia_rs                         0.1.5
lark-parser                       0.12.0
lazy_loader                       0.4
llama-cpp-agent                   0.0.17
llama_cpp_python                  0.2.26+cu121
llvmlite                          0.43.0
loguru                            0.7.2
Markdown                          3.7
markdown-it-py                    3.0.0
MarkupSafe                        2.1.5
marshmallow                       3.22.0
matplotlib                        3.9.2
matrix-client                     0.4.0
mdurl                             0.1.2
mediapipe                         0.10.14
mergedeep                         1.3.4
mkdocs                            1.6.1
mkdocs-autorefs                   1.2.0
mkdocs-get-deps                   0.2.0
mkdocs-material                   9.5.38
mkdocs-material-extensions        1.3.1
mkdocstrings                      0.26.1
mkdocstrings-python               1.11.1
mkl                               2021.4.0
ml_dtypes                         0.5.0
moviepy                           1.0.3
mpmath                            1.3.0
mss                               9.0.2
multidict                         6.0.5
multiprocess                      0.70.16
mypy-extensions                   1.0.0
networkx                          3.3
numba                             0.60.0
numpy                             1.26.4
oauthlib                          3.2.2
omegaconf                         2.3.0
onnxruntime                       1.19.2
onnxruntime-gpu                   1.15.1
open_clip_torch                   2.26.1
openai                            1.50.0
opencv-contrib-python             4.10.0.84
opencv-python                     4.10.0.84
opencv-python-headless            4.10.0.84
opt_einsum                        3.4.0
packaging                         24.1
paginate                          0.5.7
pandas                            2.2.3
pathspec                          0.12.1
peft                              0.13.0
pendulum                          3.0.0
piexif                            1.1.3
pilgram                           1.2.1
pillow                            10.4.0
pip                               24.2
platformdirs                      4.3.6
pluggy                            1.5.0
pooch                             1.8.2
prettytable                       3.11.0
proglog                           0.1.10
prometheus_client                 0.21.0
prometheus-fastapi-instrumentator 6.0.0
proto-plus                        1.24.0
protobuf                          5.28.2
psutil                            6.0.0
PuLP                              2.9.0
py-cpuinfo                        9.0.0
pyarrow                           17.0.0
pyasn1                            0.6.1
pyasn1_modules                    0.4.1
pybase64                          1.4.0
pycparser                         2.22
pydantic                          2.9.2
pydantic_core                     2.23.4
pydantic-settings                 2.5.2
pydot                             3.0.2
pygit2                            1.15.1
PyGithub                          2.3.0
Pygments                          2.18.0
PyJWT                             2.8.0
PyMatting                         1.1.12
pymdown-extensions                10.10.2
PyNaCl                            1.5.0
pyparsing                         3.1.4
pypng                             0.20220715.0
pyreadline3                       3.5.4
pySmartDL                         1.3.4
PySocks                           1.7.1
pytest                            8.3.3
pytest-asyncio                    0.21.1
python-dateutil                   2.9.0.post0
python-dotenv                     1.0.1
python-slugify                    8.0.4
pytz                              2024.2
PyWavelets                        1.7.0
pywin32                           306
PyYAML                            6.0.2
pyyaml_env_tag                    0.1
pyzbar                            0.1.9
qrcode                            7.4.2
redis                             5.0.8
referencing                       0.35.1
regex                             2024.7.24
rembg                             2.0.59
repath                            0.9.0
requests                          2.31.0
requests-file                     2.1.0
requests-toolbelt                 1.0.0
requirements-parser               0.11.0
rf-clip                           1.1
rich                              13.5.2
rich-argparse                     1.5.2
rouge                             1.0.1
rpds-py                           0.20.0
rsa                               4.7.2
s3transfer                        0.10.2
safetensors                       0.4.5
scikit-build                      0.18.1
scikit-image                      0.24.0
scikit-learn                      1.5.2
scipy                             1.14.1
seaborn                           0.13.2
segment-anything                  1.0
sentencepiece                     0.2.0
sentry-sdk                        2.14.0
setproctitle                      1.3.3
setuptools                        71.1.0
shapely                           2.0.6
shellingham                       1.5.4
simpleeval                        0.9.13
six                               1.16.0
skypilot                          0.5.0
smmap                             5.0.1
sniffio                           1.3.1
sounddevice                       0.5.0
soundfile                         0.12.1
soupsieve                         2.6
spandrel                          0.4.0
starlette                         0.37.2
structlog                         24.4.0
supervision                       0.22.0
surrealist                        1.0.0
sympy                             1.13.1
symusic                           0.5.3
tabulate                          0.9.0
tbb                               2021.13.0
text-unidecode                    1.3
threadpoolctl                     3.5.0
tifffile                          2024.9.20
time-machine                      2.15.0
timm                              0.6.13
tldextract                        5.1.2
tokenizers                        0.20.0
tomli                             2.0.1
torch                             2.4.1+cu121
torchaudio                        2.4.1+cu121
torchscale                        0.3.0
torchsde                          0.2.6
torchvision                       0.19.1+cu121
tqdm                              4.66.5
trampoline                        0.1.2
transformers                      4.45.1
transparent-background            1.3.2
typer                             0.9.0
typer-config                      1.4.0
types-python-dateutil             2.9.0.20240906
types-setuptools                  75.1.0.20240917
typing                            3.7.4.3
typing_extensions                 4.12.2
typing-inspect                    0.9.0
tzdata                            2024.2
tzlocal                           5.2
ultralytics                       8.2.102
ultralytics-thop                  2.0.8
uritemplate                       4.1.1
urllib3                           1.26.19
uvicorn                           0.31.0
wandb                             0.18.2
watchdog                          4.0.2
watchfiles                        0.24.0
wcwidth                           0.2.13
websocket-client                  1.8.0
websockets                        13.1
wget                              3.2
wheel                             0.43.0
win32-setctime                    1.1.0
wrapt                             1.16.0
xxhash                            3.5.0
yapf                              0.40.2
yarl                              1.13.1
zipp                              3.20.2
zxing-cpp                         2.2.0
`

**Comment by kijai** [comment]
Diffusers version does look correct indeed.

How exactly did you download the model? I can't understand how this could happen if the auto download was used, it's almost as if the model is simply a wrong one. The state_dict of the I2V model does not have `context_embedder`, none of the CogVideoX models do.

**Comment by Kh4rj1** [comment]
I think I figured it out, for whatever reason the files in \ComfyUI_windows_portable\ComfyUI\models\CogVideo\CogVideoX-5b-I2V were wrong / not complete, and I did do some tampering there which I didn't fully understand on the advice of chatgpt (I don't wanna talk about it, lol)

I just ended up deleting the contents of the folder and doing `git pull https://huggingface.co/THUDM/CogVideoX-5b-I2V` , let that run overnight and now the error is gone.

Sorry for this confusing issue, I appreciate the help!
